### PR TITLE
Remove multiple parsing of addon.xml

### PR DESCRIPTION
**Pull request by mzfr**
We were parsing same addon.xml file multiple time. so I have made changes to parse the file only at one place and then pass that parsed file as a parameter to all the function.

### CODE REVIEW

File: kodi_addon_checker/check_addon.py [hunk 1]
```diff
@@ -356,22 +352,18 @@ def _get_addons(xml_url):
...
-
     return {
         i.get("addon"): i.get("version")
         for i in tree.findall("requires/import")
     }
 
 
-def _check_dependencies(report: Report, addon_path, repo_addons):
-    deps = _get_users_dependencies(addon_path)
+def _check_dependencies(report: Report, addon_path, repo_addons, tree):
```
[the remark this answers is not in the document]
Comment (mzfr) [reply kodi_addon_checker/check_addon.py]: I have fixed it

**Comment by mzfr** [comment]
@Razzeee I have resolved the conflicts. Also I changed logger variable to uppercase

File: kodi_addon_checker/check_addon.py [hunk 2]
```diff
@@ -155,16 +157,15 @@ def _check_for_invalid_json_files(report: Report, file_index):
...
-def _check_addon_xml(report: Report, addon_path):
+def _check_addon_xml(report: Report, addon_path, addon_xml):
     addon_xml_path = os.path.join(addon_path, "addon.xml")
-    addon_xml = None
     try:
         _addon_file_exists(report, addon_path, r"addon\.xml")
 
-        addon_xml = ET.parse(addon_xml_path)
-        addon = addon_xml.getroot()
+        # addon_xml = ET.parse(addon_xml_path)
```
Comment (Razzeee) [inline kodi_addon_checker/check_addon.py]: Remove these, don't comment them out

File: kodi_addon_checker/check_addon.py [hunk 3]
```diff
@@ -155,16 +157,15 @@ def _check_for_invalid_json_files(report: Report, file_index):
                                   relative_path(path)))
 
 
-def _check_addon_xml(report: Report, addon_path):
+def _check_addon_xml(report: Report, addon_path, addon_xml):
```
Comment (Razzeee) [inline kodi_addon_checker/check_addon.py]: Let's keep the name consistent, either tree or addon_xml. Or go for something that better describes it, like xml_root?

Comment (mzfr) [reply kodi_addon_checker/check_addon.py]: how about parsed_xml ?

Comment (mzfr) [reply kodi_addon_checker/check_addon.py]: Fixed

File: kodi_addon_checker/check_addon.py (hunk 2 above)
Comment (mzfr) [reply kodi_addon_checker/check_addon.py]: Fixed

**Pull request merged**